### ISSUE TITLE
Wrong date conversion if a date is "round" day

### ISSUE THREAD

**Issue by korjavin** [ClickHouse/clickhouse-go#234]
In my code I have a select 
```
var a,b time.Time
...
...QueryContext(ctx, "select .. where ts >= ? and ts< ?", a,b)
```

It works well most of the time, but if `a` or `b` is exact day I see that sql query in clickhouse-log is incorect
it becomes 
```
ts >= toDate(18164) AND ts < toDate(18222)
```
(toDate instead of toDateTime)

and I'm getting empty result

**Comment by Loofort** [comment]
I've faced the same issue.
My table has a field f type of DateTime.
The go's time.Time parameter is converted to toDate(timestamp_in_days) if time.time is the exact day. and a query fails because of a type mismatching. 
As a workaround I'm using
`f > toDateTime(?)`

**Comment by eqinox76** [comment]
We see the same issue when using 
```go
startTime := time.Now()
endTime := time.Date(startTime.Year(), startTime.Month(), startTime.Day(), 0, 0, 0, 0, startTime.Location()).AddDate(0,1,0)
rows, err := conn.Query(
		`SELECT *
		FROM myprices p 
		WHERE timestamp BETWEEN ? AND ?
		ORDER BY timestamp`,
		startTime,
		endTime,
```
the clickhouse driver converts one parameter to a Date which cannot be compared to a DateTime:
```
[clickhouse][connect=1][prepare] SELECT *
		FROM myprices p
		WHERE timestamp BETWEEN ? AND ?
		ORDER BY timestamp
[clickhouse][connect=1][send query] SELECT *
		FROM myprices p
		WHERE timestamp BETWEEN toDateTime(1611093600) AND toDate(18649)
		ORDER BY timestamp
[clickhouse][connect=1][send external tables] count 0
```

Thanks for this cool library and especially the debug feature!

**Comment by kshvakov** [comment]
Fixed in [v1.5.2](https://github.com/ClickHouse/clickhouse-go/releases/tag/v1.5.2)
